homeassistant/components/nibe_heatpump/sensor.py

```diff
@@ -0,0 +1,79 @@
+"""The Nibe Heat Pump sensors."""
+from __future__ import annotations
+
+from nibe.coil import Coil
+
+from homeassistant.components.sensor import (
+    ENTITY_ID_FORMAT,
+    SensorDeviceClass,
+    SensorEntity,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_CURRENT_MILLIAMPERE,
+    ELECTRIC_POTENTIAL_MILLIVOLT,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    ENERGY_MEGA_WATT_HOUR,
+    ENERGY_WATT_HOUR,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+    TEMP_KELVIN,
+    TIME_HOURS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import DOMAIN, CoilEntity, Coordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up platform."""
+
+    coordinator: Coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        [
+            Sensor(coordinator, coil)
+            for coil in coordinator.coils
+            if not coil.is_writable and not coil.is_boolean
+        ]
```

Comment:
```suggestion
            Sensor(coordinator, coil)
            for coil in coordinator.coils
            if not coil.is_writable and not coil.is_boolean
```

This could be a generator expression